engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -543,61 +560,82 @@ case class WorkflowExecutionActor(workflowDescriptor: EngineWorkflowDescriptor,
-//          s"Scatter collection must evaluate to an array but instead got ${v.wdlType.toWdlString}")
-//      }
-//    }
-    Failure(new Exception("BOOM"))
-  }
-
-  private def processRunnableCollector(collector: CollectorKey, data: WorkflowExecutionActorData, isInBypassed: Boolean): Try[WorkflowExecutionDiff] = {
-
-    val shards = data.executionStore.findCompletedShardsForOutput(collector)
-
-    data.outputStore.generateCollectorOutput(collector, shards) match {
-      case Failure(e) => Failure(new RuntimeException(s"Failed to collect output shards for call ${collector.tag}", e))
-      case Success(outputs) =>
-        val adjustedOutputs: CallOutputs = if (isInBypassed) {
-          outputs map { output => (output._1, JobOutput(WdlOptionalValue.none(output._2.wdlValue.wdlType) )) }
-        } else outputs
-        self ! ScatterCollectionSucceededResponse(collector, adjustedOutputs)
-        Success(WorkflowExecutionDiff(Map(collector -> ExecutionStatus.Starting)))
+    if (bypassed) {
+      Success(WorkflowExecutionDiff(scatterKey.populate(0) + (scatterKey -> ExecutionStatus.Bypassed)))
+    } else {
+      val collectionOutputPort = scatterKey.node.scatterCollectionExpressionNode.singleExpressionOutputPort
+
+      data.valueStore.get(collectionOutputPort, None) map {
+        case WdlArrayLike(arrayLike) =>
+          Success(
+            WorkflowExecutionDiff(
+              // Add the new shards + collectors, and set the scatterVariable and scatterKey to Done
+              executionStoreChanges = scatterKey.populate(arrayLike.value.size) ++ Map (
+                ScatterVariableInputKey(scatterKey.node.scatterVariableInnerGraphInputNode, arrayLike) -> Done,
+                scatterKey -> Done
+              ),
+              // Add scatter variable arrayLike to the value store
+              valueStoreAdditions = Map (
+                OutputKey(scatterKey.node.scatterVariableInnerGraphInputNode.singleOutputPort, None) -> arrayLike
+              )
+            )
+          )
+        case v: WdlValue =>
+          Failure(new RuntimeException(s"Scatter collection must evaluate to an array but instead got ${v.wdlType.toWdlString}"))
+      } getOrElse {
+        Failure(new RuntimeException(s"Could not find an array value for scatter ${scatterKey.tag}"))
+      }
+    }
+  }
+
+  /**
+    * Collects all shards and add them to the value store
+    */
+  private def processRunnableScatterCollector(collector: ScatterCollectorKey, data: WorkflowExecutionActorData, isInBypassed: Boolean): Try[WorkflowExecutionDiff] = {
+    data.valueStore.collectShards(collector) match {
+      case Invalid(e) => Failure(new RuntimeException with MessageAggregation {
+        override def exceptionContext: String = s"Failed to collect output shards for node ${collector.tag}"
+        override def errorMessages: Traversable[String] = e.toList
+      })
+      case Valid(outputs) =>
+        // TODO WOM: fix
```

Comment:
This was bit was previously there to adjust the values if they were in a bypassed conditional. I'm not sure it will still be needed when conditional work but I left it in case. It doesn't hurt scatter without conditionals